test/over_react/component/ref_util_test.dart

```diff
@@ -0,0 +1,332 @@
+        test('that are in a List literal', () {
+          expect(() => mount(getFactoryForDiv()()(['test'])), returnsNormally);
+        });
+
+        test('that are not in a List literal', () {
+          expect(() => mount(getFactoryForDiv()()('test')), returnsNormally);
+        });
+      });
+    });
+
+    group('on a component with a dart component child', () {
+      testForwardRefWith(Basic, verifyRefValue: (ref) {
+        expect(ref, TypeMatcher<BasicComponent>());
+      }, useUiForwardRef: useUiForwardRef);
+
+      group('- sets displayName on the rendered component as expected', () {
+        test('when displayName argument is not passed to forwardRef', () {
+          final BasicForwarded = getFactoryForBasic();
+
+          final Ref refObject = createRef();
+          final vDomElement = (BasicForwarded()..ref = refObject)();
+
+          expect(getProperty(getProperty(vDomElement.type, 'render'), 'displayName'),
+              useUiForwardRef ? 'Anonymous' : 'Basic');
+        });
+
+        test('when displayName argument is passed to the config constructor', () {
+          final BasicForwarded = getFactoryForBasic(displayName: displayName);
+
+          final Ref refObject = createRef();
+          final vDomElement = (BasicForwarded()..ref = refObject)();
+
+          expect(getProperty(getProperty(vDomElement.type, 'render'), 'displayName'), displayName);
+        });
+      });
+
+      group('returns normally when passed children', () {
+        test('that are in a List literal', () {
+          expect(() => mount(getFactoryForBasic()()(['test'])), returnsNormally);
+        });
+
+        test('that are not in a List literal', () {
+          expect(() => mount(getFactoryForBasic()()('test')), returnsNormally);
+        });
+      });
+    });
+  });
+}
+
+const displayName = 'AVerySpecificDisplayName';
+
+void testForwardRefWith(dynamic factory,
+    {void Function(dynamic refValue) verifyRefValue, useUiForwardRef = false}) {
+  test('- passes a ref through the parent to its child', () {
+    UiFactory<BasicProps> BasicForwarded = useUiForwardRef
+        ? uiForwardRef<BasicProps>((props, ref) {
+            return (factory()
+              ..ref = ref
+              ..id = props.childId)();
+          }, Basic.asForwardRefConfig())
+        : forwardRef<BasicProps>((props, ref) {
+            return (factory()
+              ..ref = ref
+              ..id = props.childId)();
+          })(Basic);
+
+    final Ref refObject = createRef();
+
+    mount((BasicForwarded()
+      ..ref = refObject
+      ..childId = 'test')());
+
+    // component props are accessed differently depending on if it is a dom component
+    // or a dart component
+    String idValue;
+    if (refObject.current is Element) {
+      idValue = refObject.current.id;
+    } else {
+      idValue = refObject.current.props['id'];
+    }
+
+    expect(idValue, equals('test'), reason: 'child component should have access to parent props');
```

Comment:
This reason is a little bit misleading; what about something like "child component receives props passed to it"